src/auth/test-libpassword.c

```diff
@@ -111,7 +111,7 @@ static void test_password_schemes(void)
 	test_password_scheme("SCRAM-SHA-1", "{SCRAM-SHA-1}4096,GetyLXdBuHzf1FWf8SLz2Q==,NA/OqmF4hhrsrB9KR7po+dliTGM=,QBiURvQaE6H6qYTmeghDHLANBFQ=", "test");
 	test_password_scheme("BLF-CRYPT", "{BLF-CRYPT}$2y$05$11ipvo5dR6CwkzwmhwM26OXgzXwhV2PyPuLV.Qi31ILcRcThQpEiW", "test");
 #ifdef HAVE_LIBSODIUM
-	test_password_scheme("ARGON2", "{ARGON2}$argon2i$v=19$m=32768,t=4,p=1$f2iuP4aUeNMrgu34fhOkkg$1XSZZMWlIs0zmE+snlUIcLADO3GXbA2O/hsQmmc317k", "test");
+	test_password_scheme("ARGON2I", "{ARGON2I}$argon2i$v=19$m=32768,t=4,p=1$f2iuP4aUeNMrgu34fhOkkg$1XSZZMWlIs0zmE+snlUIcLADO3GXbA2O/hsQmmc317k", "test");
```

Comment:
maybe ARGON2 should default to the "whatever is best for the platform" implementation?

Reply:
Even though there are no plans to change this in a near future, it may not be an Argon2 variant forever.

Reply:
I just wonder if there should be {ARGON2} that defaults to whatever argon wants to spew out.

Reply:
But a different variants will take a different amount of time to complete given the same CPU work factor (`OPSLIMIT`). So I'd rather use an explicit algorithm name.

Reply:
It is not necessarely a problem. We already have {CRYPT} which eats anything from des, md5, sha and bcrypt.

Reply:
Maybe call it {SODIUM} instead of {ARGON} then?

Reply:
That would not make sense. I ment that maybe add {ARGON} along with {ARGON2I} and {ARGON2ID}? sodium is a library, not a password scheme.

Reply:
Yes, but `pwhash_verify()` can verify any algorithm produced by high-level API of the library, not specifically `Argon2*`. In both cases, having `{ARGON}` or `{SODIUM}` or whatever generic thing would allow verification, but not hashing, as the parameters are not compatible between functions.

Reply:
Ok, thank you for clarifying this. Then we proceed with this.